### ISSUE TITLE
Show better where street A becomes street B

### ISSUE THREAD

**Issue by jidanni** [gravitystorm/openstreetmap-carto#4742]
Like #4741, additional ambiguity occurs at
https://www.openstreetmap.org/?mlat=42.15576&mlon=-87.82107#map=15/42.1594/-87.8279

Even at the highest zoom level, 19, we can't tell _exactly_ where Old Briar Road ends, and Magnolia Avenue begins.

[Surprise](https://www.openstreetmap.org/way/703045537#map=19/42.15584/-87.82099)!

[Google](https://www.google.com/maps/@42.15584,-87.82099,19z) does a little better, but ugly:.

[Bing](https://www.bing.com/maps?cp=42.15584~-87.82099&lvl=19) seems to conquer the problem, but not in the tiles, but by the names one sees moving the map around.

**Comment by imagico** [comment]
That would require a rather complex join in SQL to implement.  Independent of that it seems doubtful that this can be visualized in a way that is not likely to be mistaken as a mapped real world feature (like some kind of barrier, crossing, speed bump, etc).

If anyone wants to develop a workable solution to this (workable both in terms of query logic and in terms of styling) that would be be worth looking at IMO. It would still be a matter of weighing added code complexity against the benefit.

**Comment by jidanni** [comment]
Yes it seems the only solution would be back to back names.  (Not necessary in cases where the name change could be inferred from a [bend in straight streets](https://www.openstreetmap.org/?mlat=42.1564637&mlon=-87.7769643#map=17/42.1564637/-87.7769643).)
